### PR TITLE
fix dynamic states

### PR DESCRIPTION
**Pull request by Mee-gu**

### CODE REVIEW

File: cocos/renderer/core/gfx/GFXCommandBuffer.h [hunk 1]
```diff
@@ -32,7 +32,8 @@ class CC_CORE_API GFXCommandBuffer : public GFXObject {
   virtual void updateBuffer(GFXBuffer* buff, void* data, uint size, uint offset = 0) = 0;
   virtual void copyBufferToTexture(GFXBuffer* src, GFXTexture* dst, GFXTextureLayout layout, const GFXBufferTextureCopyList& regions) = 0;
   virtual void execute(const std::vector<GFXCommandBuffer*>& cmd_buffs, uint32_t count) = 0;
-  
+  virtual void enableDepthBias(bool enabled) { _depthBiasEnabled = enabled; }
```
Comment (minggo) [inline cocos/renderer/core/gfx/GFXCommandBuffer.h:35]: I think use setDepthBias is better.

File: cocos/renderer/gfx-gles2/GLES2Commands.cc [hunk 2]
```diff
@@ -1899,10 +1899,149 @@ void GLES2CmdFuncExecuteCmds(GLES2Device* device, GLES2CmdPackage* cmd_package)
...
+                        glPolygonOffset(cmd->depthBias.constant, cmd->depthBias.slope);
+                        cache->rs.depthBias = cmd->depthBias.constant;
+                        cache->rs.depthBiasSlop = cmd->depthBias.slope;
+                    }
+                    break;
+                case GFXDynamicState::BLEND_CONSTANTS:
+                    if ((cache->bs.blendColor.r != gpuPipelineState->bs.blendColor.r) ||
+                        (cache->bs.blendColor.g != gpuPipelineState->bs.blendColor.g) ||
+                        (cache->bs.blendColor.b != gpuPipelineState->bs.blendColor.b) ||
+                        (cache->bs.blendColor.a != gpuPipelineState->bs.blendColor.a))
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2Commands.cc:1951]: Can implement operator == and != in GFXColor to simplify the codes. And float comparison can not just use !=.

File: cocos/renderer/core/gfx/GFXCommandBuffer.h (hunk 1 above)
Comment (Mee-gu) [reply cocos/renderer/core/gfx/GFXCommandBuffer.h:35]: there is already a `setDepthBias ` to set depth bias

File: cocos/renderer/gfx-gles2/GLES2Commands.cc (hunk 2 above)
Comment (Mee-gu) [reply cocos/renderer/gfx-gles2/GLES2Commands.cc:1951]: got it.

**Comment by minggo** [comment]
I think we should support all dynamics that are supported on all platforms. As some dynamic states are on supported on metal, then can refer to molten implementation.

**Comment by Mee-gu** [comment]
> I think we should support all dynamics that are supported on all platforms. As some dynamic states are on supported on metal, then can refer to molten implementation.

I will take a look.

**Pull request merged**